spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -315,28 +314,14 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		TaskExecutionInformation taskExecutionInformation = findOrCreateTaskExecutionInformation(taskName,
 				taskDeploymentProperties, launcher.getType(), previousTaskDeploymentProperties);
 
-		// pre prosess command-line args
-		// moving things like app.<label> = arg
-		// into deployment properties if ctr and removing
-		// prefix if simple task.
 		if (taskExecutionInformation.isComposed()) {
-			List<String> composedTaskArguments = new ArrayList<>();
-			commandLineArgs.forEach(arg -> {
-				if (arg.startsWith("app.")) {
-					String[] split = arg.split("=", 2);
-					if (split.length == 2) {
-						composedTaskArguments.add("--composed-task-app-arguments." + Base64Utils.encode(split[0]) + "=" + split[1]);
-					}
-					else {
-						composedTaskArguments.add("--composed-task-app-arguments." + arg);
-					}
-				}
-				else {
-					composedTaskArguments.add(arg);
-				}
-			});
-			logger.info("composedTaskArguments {}", StringUtils.collectionToCommaDelimitedString(composedTaskArguments));
-			commandLineArgs = composedTaskArguments;
+			// pre process command-line args
+			// moving things like app.<label> = arg
+			// into deployment properties if ctr and removing
+			// prefix if simple task.
+			commandLineArgs = TaskServiceUtils.convertCommandLineArgsToCTRFormat(commandLineArgs);
+			logger.info("composedTaskArguments {}", StringUtils.collectionToCommaDelimitedString(commandLineArgs));
```

Comment:
**[TinyNit]** I have found it safer to just print out the collection rather than attempt to format the content as things could go awry in the `StringUtils.collectionTo...`. Also this gets evaluated regardless if INFO logging is off.

Reply:
Looking at this again.  I think it is bad practice to output this data at all in the log.    I willremove it.

Reply:
Good point